### DISCUSSION TITLE
[Bug] fieldModifier doesn't work properly when concat'ed with string

### DISCUSSION THREAD

**Discussion by caasion** [mdelobelle/metadatamenu#51] [Tips and Tricks]
If I try to concat a fieldModifier field with some string, it ends up in the wrong place: outside the table.

![image](https://user-images.githubusercontent.com/103465188/184393144-7280a14c-9c0e-4ef8-b64c-745fde7e6fc0.png)

Query:
```dataviewjs
const {fieldModifier: f} = this.app.plugins.plugins["metadata-menu"].api;

dv.table(["Field"], 
    await Promise.all([dv.current()]
        .map(async p => [
            await f(dv, p, "field1"),
            "Field 2:" + await f(dv, p, "field2")
        ])
    )
)
```

**Comment by mdelobelle** [comment]
Hi this is not a bug, this is due to the fact that fieldModifier returns an HTMLElement (this is what is expected from the dv.table <element> argument).
You can't concat an HTMLElement with a string

one way to achieve what you want is to create a concat function that will return an htmlElement containing 2 children:
- your string as a span Element 
- your field

example:

```javascript
const {fieldModifier: f} = this.app.plugins.plugins["metadata-menu"].api;

const concat = (text, field) => {
	const wrapper = dv.el("div", "");
	wrapper.setAttr("style", "display:flex");
	const prefix = document.createElement("span");
	prefix.setAttr("style", "margin-right: 1em");
	prefix.setText(text);
	wrapper.appendChild(prefix);
	wrapper.appendChild(field);
	return wrapper
}

dv.table(["Field1", "Field2"], 
    await Promise.all([dv.current()]
        .map(async p => [
            await f(dv, p, "field1"),
            concat("Field 2:", await f(dv, p, "field2"))
        ])
    )
)
```

**Reply by YIRU69** [reply]
@mdelobelle thanks! 
Why not the `concat(await f(dv, p, "field2"), "-Field 2")` work?
